core/domain/acl_decorators_test.py

```diff
@@ -1024,3 +1026,53 @@ def test_exploration_editor_can_access_learner_dashboard(self):
         response = self.testapp.get('/mock/', expect_errors=True)
         self.assertEqual(response.status_int, 200)
         self.logout()
+
+
+class EditTopicDecoratorTest(test_utils.GenericTestBase):
+    """Tests the decorator can_edit_topic."""
+    username1 = 'topicmanager'
+    user_email1 = 'user@example.com'
+    username2 = 'normaluser'
+    user_email2 = 'user2@example.com'
+    topic_id = 'topic_1'
+
+    class MockHandler(base.BaseHandler):
+        @acl_decorators.can_edit_topic
+        def get(self, topic_id):
+            self.render_json({'topic_id': topic_id})
+
+    def setUp(self):
+        super(EditTopicDecoratorTest, self).setUp()
+        self.signup(self.ADMIN_EMAIL, self.ADMIN_USERNAME)
+        self.signup(self.user_email1, self.username1)
+        self.signup(self.user_email2, self.username2)
+        self.set_admins([self.ADMIN_USERNAME])
+        self.set_topic_managers([self.username1])
+
+        self.admin_id = self.get_user_id_from_email(self.ADMIN_EMAIL)
+        self.manager_id = self.get_user_id_from_email(self.user_email1)
+        self.admin = user_services.UserActionsInfo(self.admin_id)
+        self.manager = user_services.UserActionsInfo(self.manager_id)
+
+        self.testapp = webtest.TestApp(webapp2.WSGIApplication(
+            [webapp2.Route('/mock/<topic_id>', self.MockHandler)],
+            debug=feconf.DEBUG,
+        ))
+        topic_services.create_new_topic_rights(self.topic_id, self.admin_id)
+        topic_services.assign_role(
+            self.admin, self.manager_id, topic_domain.ROLE_MANAGER,
+            self.topic_id)
+
+    def test_topic_manager_can_edit_topic(self):
+        self.login(self.user_email1)
+        response = self.testapp.get(
+            '/mock/%s' % self.topic_id, expect_errors=True)
```

Comment:
Why expect_errors=True?

Reply:
From the [testapp docs](https://docs.pylonsproject.org/projects/pylons-webframework/en/latest/thirdparty/webtest.html), any response code except 2xx or 3xx will throw an error (I tried this out without this parameter and the test fails in cases where we expect a 4xx response code).

Reply:
But the response code here is 200?

Reply:
That's true. I just kept the parameter true even when the expected response is 200 as then the test would error our with an assertion error rather than a bad response error. Should I remove the expect_error parameter when the expected status code is valid?

Reply:
I think so? But better to use self.get_json() if you can.

Reply:
Ok. I'll make the changes. Thanks!

Reply:
Changed this